core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala

```diff
@@ -360,18 +531,30 @@ class ConsumerBounceTest extends BaseRequestTest with Logging {
       consumer.assign(Collections.singleton(tp))
     else
       consumer.subscribe(Collections.singleton(topic))
-    receiveRecords(consumer, numRecords)
+    receiveExactRecords(consumer, numRecords)
     consumer
   }
 
-  private def receiveRecords(consumer: KafkaConsumer[Array[Byte], Array[Byte]], numRecords: Int, topic: String = this.topic, timeoutMs: Long = 60000) {
+  private def receiveRecords(consumer: KafkaConsumer[Array[Byte], Array[Byte]], numRecords: Int, timeoutMs: Long = 60000): Long = {
     var received = 0L
     val endTimeMs = System.currentTimeMillis + timeoutMs
     while (received < numRecords && System.currentTimeMillis < endTimeMs)
-      received += consumer.poll(1000).count()
+      received += consumer.poll(time.Duration.ofMillis(100)).count()
+
+    received
+  }
+
+  private def receiveExactRecords(consumer: KafkaConsumer[Array[Byte], Array[Byte]], numRecords: Int, timeoutMs: Long = 60000): Unit = {
+    val received = receiveRecords(consumer, numRecords, timeoutMs)
     assertEquals(numRecords, received)
   }
 
+  private def receiveAtLeastRecords(consumer: KafkaConsumer[Array[Byte], Array[Byte]], numRecords: Int, timeoutMs: Long): Unit = {
```

Comment:
Maybe we could call this `receiveAndCommit` or something so that the offset commit expectation is clear.